server/infrastructure/player_repository.py

```
@@ -0,0 +1,56 @@
+        self._players_cache: List[Dict] = []
+        self._cache_loaded = False
+    
+    def _load_database(self):
+        """Load all players from Firebase into memory for fast searching"""
+        if not self._cache_loaded and self.db:
+            try:
+                players_ref = self.db.collection('players').stream()
+                self._players_cache = [doc.to_dict() for doc in players_ref]
+                self._cache_loaded = True
+                print(f"Loaded {len(self._players_cache)} players from Firebase")
+            except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)